### PR TITLE
Fix webdataset python index creation script

### PR DESCRIPTION
**Pull request by JanuszL**
- fixes wrong offset calculation in webdataset python
  index creation script. The right formula for the data
  offset calculation should be the output of (`block-number` + 1) * 512
  while it was calculated as (next ``block-number`) * 512 - data size
  (what doesn't work when the first entry is preceded by additional data
  like in POSIX.1-2001 archive

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:
**Bug fix** (*non-breaking change which fixes an issue*)

## Description:.
- fixes wrong offset calculation in webdataset python
  index creation script. The right formula for the data
  offset calculation should be the output of (`block-number` + 1) * 512
  while it was calculated as (next ``block-number`) * 512 - data size
  (what doesn't work when the first entry is preceded by additional data
  like in POSIX.1-2001 archive

### Affected modules and functionalities:
- tools/webdataset index creation script

### Key points relevant for the review:
- NA

### Tests:

- [x] Existing tests apply
  - [X]
    - test_webdataset_requirements.py (but I don't know how to generate tar file with non-0 offset for the initial entry)
- [ ] New tests added
  - [ ] Python tests
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

Relates to https://github.com/NVIDIA/DALI/issues/4215

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: tools/wds2idx.py [hunk 1]
```diff
@@ -100,7 +100,7 @@ def _get_data_tar(self):
         )  # <type>... <size> <date> <name>
 
         # Extracting
-        for blocks_line, next_blocks_line, types_sizes_line in zip(tar_blocks, tar_blocks[1:],
+        for blocks_line, next_blocks_line, types_sizes_line in zip(tar_blocks, tar_blocks,
```
Comment (klecki) [inline tools/wds2idx.py:103]: Do we need to zip the same thing twice? 
We should be using current blocks_line in the calculation I guess.

File: tools/wds2idx.py [hunk 2]
```diff
@@ -111,14 +111,13 @@ def _get_data_tar(self):
...
-                next_blocks_line[next_blocks_line.find(b"block") + 6 : next_blocks_line.find(b":")]) * 512  # noqa: E203, E501
+            offset = int(next_blocks_line[next_blocks_line.find(b"block") + 6:
+                                          next_blocks_line.find(b":")])
+            offset = (offset + 1) * 512
 
             size = types_sizes_line[: -len(name)]
             size = size[: size.rfind(b"-") - 8]  # "... <size> 20yy-mm-...."
             size = int(size[size.rfind(b" "):])
-            offset -= size
-            offset -= offset % 512
```
Comment (klecki) [inline tools/wds2idx.py:121]: I am not really sure what is calculated here and what is the actual layout? Are the metadata at the end, that we can backtrack by the size and lower it to the closes multiple of 512?

File: tools/wds2idx.py (hunk 1 above)
Comment (JanuszL) [reply tools/wds2idx.py:103]: Fixed

File: tools/wds2idx.py (hunk 2 above)
Comment (JanuszL) [reply tools/wds2idx.py:121]: Added a comment.

**Comment by JanuszL** [comment]
!build

**Review by awolant — APPROVED** [review]

**Pull request merged**